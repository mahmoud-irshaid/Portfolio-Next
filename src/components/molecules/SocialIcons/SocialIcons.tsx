import type { ReactNode } from 'react';

export interface SocialIconsProps {
	profiles: Array<{
		url: string;
		name: string;
		icon: ReactNode;
		mail?: boolean;
	}>;
}

const SocialIcons = ({ profiles }: SocialIconsProps) => {
	return (
		<ul className="flex gap-6">
			{profiles.map(({ url, name, icon, mail }) => (
				<li
					className="text-2xl opacity-70 transition hover:opacity-100"
					key={name}
				>
					<a href={mail ? "mailto:mahmoudirshaid10@gmail.com" : url} target="_blank" rel="noopener noreferrer" title={name}>
						{icon}
					</a>
				</li>
			))}
		</ul>
	);
};

export { SocialIcons };
