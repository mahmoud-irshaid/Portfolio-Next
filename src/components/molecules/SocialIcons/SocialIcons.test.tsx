import { SocialIcons } from '../SocialIcons';

import { render } from '@testing-library/react';

describe('SocialIcons', () => {
	it('renders correctly', () => {
		const { container } = render(
			<SocialIcons
				profiles={[
					{
						url: 'https://github.com',
						name: 'Github',
						icon: 'GITHUB',
					},
				]}
			/>
		);
		expect(container).toMatchSnapshot();
	});
});
