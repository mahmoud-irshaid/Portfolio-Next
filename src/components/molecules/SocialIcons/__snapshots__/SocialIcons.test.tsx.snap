// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialIcons renders correctly 1`] = `
<div>
  <ul
    class="flex gap-6"
  >
    <li
      class="text-2xl opacity-70 transition hover:opacity-100"
    >
      <a
        href="https://github.com"
        rel="noopener noreferrer"
        target="_blank"
        title="Github"
      >
        GITHUB
      </a>
    </li>
  </ul>
</div>
`;
